README.md

```diff
@@ -72,6 +72,7 @@ source build-docker.sh
 ```
 source docker-compose.sh up --scale mpi_head=1 --scale mpi_node=3
 docker exec -u nixuser -it waterpaths_mpi_head_1 /bin/sh
+nix-shell dev-env.nix

```

Comment:
Just want to double check if this is needed? haven't looked closely yet. Also if I recall `dev-env.nix` is not meant to be used with `nix-shell`, but rather e.g. `nix-env` .. however it has been a while and I could be remembering my own conventions incorrectly.

Reply:
It is not necessary, but is for the benefit of the user.  I am not certain what your conventions were, since you were out at the time, but it was the way that worked to give Bernardo the dev environment he was expecting (or at least closer to it).  He was really struggling with the shell environment as it was before that.  I welcome suggestions, if you have them, for how to give him a dev environment otherwise because he was quite set on it.

Reply:
In Lake_Problem_DPS, you can just use `nix-shell .` and it will give you the dev environment detailed in default.nix, but yours was set up differently.  This is actually something I was planning on addressing in our future templates.

Reply:
OK, sounds good - but I don't see the `dev-env.nix` file in the `develop` branch currently, but maybe I just missed it.

Reply:
Typically yes, I used `dev-env.nix` to set up a development environment, and another file (`shell.nix` or `default.nix`) to just be able to "run the app" in a minimal way.

Reply:
I suppose that isn't unlike Python/JS build environments where you have your core dependencies, and then your dev dependencies, specified  in two different groups, so the latter are not deployed.